File: dehaze-go/internal/service/message/message_template_service.go
package message

import (
	"context"
	"encoding/json"

	"github.com/earthyzinc/dehaze-go/internal/model/query"
	"github.com/earthyzinc/dehaze-go/internal/model/vo"
	msgrepo "github.com/earthyzinc/dehaze-go/internal/repository/message"
	"github.com/earthyzinc/dehaze-go/internal/model/bo"
	"github.com/earthyzinc/dehaze-go/pkg/common"
)

type MessageTemplateService struct {
	tplRepo msgrepo.IMessageTemplateRepository
}

func NewMessageTemplateService(tplRepo msgrepo.IMessageTemplateRepository) *MessageTemplateService {
	return &MessageTemplateService{tplRepo: tplRepo}
}

func (s *MessageTemplateService) GetPage(ctx context.Context, q *query.MessageTemplateQuery) (*vo.PageResult[vo.MessageTemplateVO], error) {
	tpls, total, err := s.tplRepo.FindPage(ctx, q)
	if err != nil {
		return nil, common.WrapBizError(common.DATABASE_ERROR, "查询模板列表失败", err)
	}
	list := make([]vo.MessageTemplateVO, 0, len(tpls))
	for _, tpl := range tpls {
		list = append(list, vo.MessageTemplateVO{
			ID:            tpl.ID,
			Code:          tpl.Code,
			Name:          tpl.Name,
			Type:          tpl.Type,
			TitleTemplate: tpl.TitleTemplate,
			Priority:      int(tpl.Priority),
			Status:        int(tpl.Status),
			CreateTime:    formatTimeVal(tpl.CreatedAt),
		})
	}
	return &vo.PageResult[vo.MessageTemplateVO]{List: list, Total: total}, nil
}

func (s *MessageTemplateService) GetDetail(ctx context.Context, id int64) (*vo.MessageTemplateDetailVO, error) {
	tpl, err := s.tplRepo.FindByID(ctx, id)
	if err != nil {
		return nil, common.WrapBizError(common.DATABASE_ERROR, "查询模板失败", err)
	}
	if tpl == nil {
		return nil, common.NewBizError(common.RESOURCE_NOT_FOUND, "模板不存在")
	}

	var channels interface{}
	if tpl.Channels != "" {
		_ = json.Unmarshal([]byte(tpl.Channels), &channels)
	}

	var variables []vo.TemplateVarVO
	if tpl.Variables != "" {
		_ = json.Unmarshal([]byte(tpl.Variables), &variables)
	}

	return &vo.MessageTemplateDetailVO{
		ID:              tpl.ID,
		Code:            tpl.Code,
		Name:            tpl.Name,
		Type:            tpl.Type,
		TitleTemplate:   tpl.TitleTemplate,
		ContentTemplate: tpl.ContentTemplate,
		Priority:        int(tpl.Priority),
		Channels:        channels,
		Variables:       variables,
		Status:          int(tpl.Status),
		CreateTime:      formatTimeVal(tpl.CreatedAt),
		UpdateTime:      formatTimeVal(tpl.UpdatedAt),
	}, nil
}

func (s *MessageTemplateService) Update(ctx context.Context, id int64, userID int64, form *bo.MessageTemplateForm) error {
	tpl, err := s.tplRepo.FindByID(ctx, id)
	if err != nil {
		return common.WrapBizError(common.DATABASE_ERROR, "查询模板失败", err)
	}
	if tpl == nil {
		return common.NewBizError(common.RESOURCE_NOT_FOUND, "模板不存在")
	}

	updates := make(map[string]interface{})
	if form.Name != "" {
		updates["name"] = form.Name
	}
	if form.TitleTemplate != "" {
		updates["title_template"] = form.TitleTemplate
	}
	if form.ContentTemplate != "" {
		updates["content_template"] = form.ContentTemplate
	}
	if form.Priority != nil {
		updates["priority"] = *form.Priority
	}
	if form.Channels != nil {
		updates["channels"] = toJSONString(form.Channels)
	}
	if form.Status != nil {
		updates["status"] = *form.Status
	}
	updates["update_by"] = userID

	if len(updates) > 1 {
		if err := s.tplRepo.Update(ctx, id, updates); err != nil {
			return common.WrapBizError(common.DATABASE_ERROR, "更新模板失败", err)
		}
	}
	return nil
}

var _ IMessageTemplateService = (*MessageTemplateService)(nil)

File: dehaze-python/app/service/message_template_service.py
from datetime import datetime
from typing import Any

from sqlalchemy.ext.asyncio import AsyncSession

from app.core.code import ResultCode
from app.core.exceptions import BusinessException
from app.repository.message_template_repository import message_template_repository


def _format_dt(dt: datetime | None) -> str | None:
    if dt is None:
        return None
    return dt.strftime("%Y-%m-%d %H:%M:%S")


class MessageTemplateService:
    async def get_page(self, 
        db: AsyncSession,
        page: int,
        page_size: int,
        name: str | None = None,
        type: str | None = None,
        status: int | None = None,
    ) -> dict[str, Any]:
        items, total = await message_template_repository.get_page(
            db, page, page_size, name, type, status
        )
        list_data = [
            {
                "id": t.id,
                "code": t.code,
                "name": t.name,
                "type": t.type,
                "titleTemplate": t.title_template,
                "priority": t.priority,
                "status": t.status,
                "createTime": _format_dt(t.create_time),
            }
            for t in items
        ]
        return {"list": list_data, "total": total, "pageNum": page, "pageSize": page_size}

    async def get_detail(self, db: AsyncSession, template_id: int) -> dict[str, Any]:
        template = await message_template_repository.get_by_id(db, template_id)
        if not template:
            raise BusinessException(ResultCode.RESOURCE_NOT_FOUND, "模板不存在")
        return {
            "id": template.id,
            "code": template.code,
            "name": template.name,
            "type": template.type,
            "titleTemplate": template.title_template,
            "contentTemplate": template.content_template,
            "priority": template.priority,
            "channels": template.channels,
            "variables": template.variables,
            "status": template.status,
            "createTime": _format_dt(template.create_time),
            "updateTime": _format_dt(template.update_time),
        }

    async def update(self, db: AsyncSession, template_id: int, data: dict[str, Any]) -> None:
        template = await message_template_repository.get_by_id(db, template_id)
        if not template:
            raise BusinessException(ResultCode.RESOURCE_NOT_FOUND, "模板不存在")

        if "name" in data:
            template.name = data["name"]
        if "titleTemplate" in data:
            template.title_template = data["titleTemplate"]
        if "contentTemplate" in data:
            template.content_template = data["contentTemplate"]
        if "priority" in data:
            template.priority = data["priority"]
        if "channels" in data:
            template.channels = data["channels"]
        if "status" in data:
            template.status = data["status"]
        await db.flush()


# 单例
message_template_service = MessageTemplateService()
